### PR TITLE
tidying & documenting control logic

### PR DESCRIPTION
**Pull request by EdHastingsCasperAssociation**
* some log tuning
* mild adjustment of sync state management
* some renaming of things for clarity
* lots of explanatory comments

### CODE REVIEW

File: node/src/reactor/main_reactor/catch_up.rs [hunk 1]
```diff
@@ -35,13 +34,16 @@ impl MainReactor {
...
+        // if there is instruction, return to start working on it
+        // else fall thru with the current best available id for block syncing
+        let sync_identifier = match self.catch_up_process() {
+            Either::Right(catch_up_instruction) => return catch_up_instruction,
+            Either::Left(sync_identifier) => sync_identifier,
         };
-        debug!("CatchUp: starting with {:?}", starting_with);
-
-        let sync_instruction = self.block_accumulator.sync_instruction(starting_with);
+        debug!("CatchUp: starting with {:?}", sync_identifier);
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/catch_up.rs]:
```suggestion
        debug!("CatchUp: sync_identifier {:?}", sync_identifier);
```

File: node/src/reactor/main_reactor/keep_up.rs [hunk 2]
```diff
@@ -64,27 +62,34 @@ impl MainReactor {
...
+        // else fall thru with the current best available id for block syncing
+        let sync_identifier = match self.keep_up_process() {
             Either::Right(keep_up_instruction) => return keep_up_instruction,
-            Either::Left(starting_with) => starting_with,
+            Either::Left(sync_identifier) => sync_identifier,
         };
         debug!(
-            ?starting_with,
+            ?sync_identifier,
             "KeepUp: starting with {}",
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/keep_up.rs]:
```suggestion
            "KeepUp: sync_identifier {}",
```

File: node/src/reactor/main_reactor/validate.rs [hunk 3]
```diff
@@ -98,7 +98,9 @@ impl MainReactor {
             None => highest_switch_block_header.height(),
         };
 
-        if !self.deploy_buffer.have_full_ttl_of_deploys(from_height) {
+        if self.deploy_buffer.have_full_ttl_of_deploys(from_height) {
+            info!("Validate: sufficient deploy TTL awareness to safely participate in consensus");
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/validate.rs]: Consider `debug!()` or `trace!()` here, unless we're sure that it's not called periodically during the normal operation of the node in KeepUp/Validate.

File: node/src/reactor/main_reactor/catch_up.rs [hunk 4]
```diff
@@ -161,7 +192,7 @@ impl MainReactor {
                     }
                     Ok(None) => {
                         // should be unreachable if we've gotten this far
-                        Either::Left(StartingWith::Hash(trusted_hash))
+                        Either::Left(SyncIdentifier::BlockHash(trusted_hash))
```
Comment (rafal-ch) [inline node/src/reactor/main_reactor/catch_up.rs]: Consider adding an error log if this is indeed expected to be unreachable.

**Review by rafal-ch — APPROVED** [review]
No blockers, just some nits.
LGTM :+1:

**Review by georgepisaltu — APPROVED** [review]

File: node/src/reactor/main_reactor/catch_up.rs (hunk 4 above)
Comment (EdHastingsCasperAssociation) [reply node/src/reactor/main_reactor/catch_up.rs]: actually, that's an old comment. I don't think it is unreachable...we could have headers locally and even block bodies, but no complete blocks. I'm removing the comment instead.

File: node/src/reactor/main_reactor/catch_up.rs (hunk 1 above)
Comment (EdHastingsCasperAssociation) [reply node/src/reactor/main_reactor/catch_up.rs]: did this

File: node/src/reactor/main_reactor/validate.rs (hunk 3 above)
Comment (EdHastingsCasperAssociation) [reply node/src/reactor/main_reactor/validate.rs]: did this

File: node/src/reactor/main_reactor/keep_up.rs (hunk 2 above)
Comment (EdHastingsCasperAssociation) [reply node/src/reactor/main_reactor/keep_up.rs]: did this

**Pull request merged**